src/main/kotlin/com/exactpro/th2/codec/EncodeProcessor.kt

```diff
@@ -50,8 +51,16 @@ class EncodeProcessor(
                 continue
             }
 
-            if (!messageGroup.isEncodable()) {
-                parentEventId.onErrorEvent("No messages of $protocols protocol or mixed empty and non-empty protocols are present", messageGroup.messageIds)
+            val msgProtocols = messageGroup.protocols
+            if (msgProtocols.none(String::isBlank) && protocols.none { it in msgProtocols }) {
+                logger.debug { "Messages with $msgProtocols protocols instead of $protocols are presented" }
+                messageBatch.addGroups(messageGroup)
+                continue
+            } else if (msgProtocols.all(String::isBlank) || msgProtocols.none(String::isBlank) && protocols.any { it in msgProtocols }) {
+                // do nothing
+            } else {
+                val info = "No messages of $protocols protocols or mixed empty and non-empty protocols are present"
+                parentEventId.onErrorEvent(info, messageGroup.messageIds)
```

Comment:
```suggestion
            when { 
              protocols.none { it.isBlank() || it in msgProtocols } -> {
                  logger.debug { "Messages with $msgProtocols protocols instead of $protocols are presented" }
                  messageBatch.addGroups(messageGroup)
                  continue
              } 
               msgProtocols.all(String::isBlank) || msgProtocols.none(String::isBlank) && protocols.any { it in msgProtocols } -> {
                  // do nothing
              } 
              else -> {
                  val info = "No messages of $protocols protocols or mixed empty and non-empty protocols are present"
                  parentEventId.onErrorEvent(info, messageGroup.messageIds)
                   continue
              }
```

Reply:
also probably we can remove "do nothing" and else blocks for replacing with only one case:
`!msgProtocols.all(String::isBlank) && (!msgProtocols.none(String::isBlank) || !protocols.any { it in msgProtocols })`

But im not sure if it's representative for people

Reply:
All above same for decode processor

Reply:
Also we can extract method with name like checkaAgainst

```
try {
    if (currentProtocols.checkProtocols(protocols)) {
        codec.encode(messageGroup).let { ... }
    } else {
       logger.debug { "Messages with $msgProtocols protocols instead of $protocols are presented" }
    }
} catch {
    // on check we also throw error of mixing empty and non empty group
    parentEventId.onErrorEvent("Failed to encode message group", messageGroup.messageIds, it)
}

```

With this we will remove excess conditions and just process data with pure logic

Reply:
I also agree that protocol check should be extracted into a separate method